### PR TITLE
Adding create, update & delete to class to avoid xmlrunner to overwirte xml file

### PR DESCRIPTION
**Pull request by ericbsd**

### CODE REVIEW

**Comment by pkgdemon** [comment]
@william-gr In addition to @ericbsd removing the build files middlewared.client.egg-info directory to .gitignore?

**Comment by william-gr** [comment]
@pkgdemon yes, we can do that

**Comment by ericbsd** [comment]
I did add the change

File: src/middlewared/middlewared/pytest/target.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [Target]
-hostname = 192.168.100.208
+hostname = 192.168.2.19
 api = /api/v2.0/
-username = root
-password = freenas
+username = "root"
+password = "testing"
```
Comment (william-gr) [inline src/middlewared/middlewared/pytest/target.conf]: I believe these should be without quote. @pkgdemon would know for sure.

Comment (ericbsd) [reply src/middlewared/middlewared/pytest/target.conf]: Oh, My fault that file was not supposed to get committed.

**Comment by william-gr** [comment]
@ericbsd now you have removed the file, as opposed to discarding the changes you made.

File: src/middlewared/middlewared/pytest/target.conf (hunk 1 above)
Comment (pkgdemon) [reply src/middlewared/middlewared/pytest/target.conf]: Yes this should be without quotes @ericbsd

**Comment by ericbsd** [comment]
Sorry about that william-rg

**Comment by pkgdemon** [comment]
@ericbsd After your cleanup let me run this through middleware, and api jobs in jenkins first before we merge also.

**Comment by ericbsd** [comment]
I will make another clone to change my machine to test my change to avoid to commit a change.

**Comment by william-gr** [comment]
Lets use "Squash and merge" github option also please so we don't pollute history, just an OCD thing I have lol : )

**Comment by william-gr** [comment]
rebuild

I cancelled this task by mistake

**Comment by william-gr** [comment]
rebuild

**Comment by pkgdemon** [comment]
ReST API tests results look good:

https://builds.ixsystems.com/jenkins/view/QA%20Tests/view/All/job/FreeNAS%20-%20Master%20-%20Incremental%20Build-%20ReST%20API%20Tests/25/testReport/(root)/

**Comment by pkgdemon** [comment]
Middleware results are the same so nothing has regressed:

https://builds.ixsystems.com/jenkins/view/QA%20Tests/view/All/job/FreeNAS%20-%20Master%20-%20Incremental%20Build-%20Middlewared%20Tests/12/testReport/

File: src/middlewared/middlewared/pytest/target.conf [hunk 2]
```diff
@@ -1,4 +1,3 @@
-[Target]
```
Comment (william-gr) [inline src/middlewared/middlewared/pytest/target.conf]: I think that line was removed by mistake as well?

Comment (ericbsd) [reply src/middlewared/middlewared/pytest/target.conf]: Oh Yeah, I forgot to add it back thanks @william-gr

**Pull request merged**